Source/ThirdParty/skia/CMakeLists.txt

```diff
@@ -0,0 +1,1019 @@
+    src/sksl/ir/SkSLIndexExpression.cpp
+    src/sksl/ir/SkSLInterfaceBlock.cpp
+    src/sksl/ir/SkSLLayout.cpp
+    src/sksl/ir/SkSLLiteral.cpp
+    src/sksl/ir/SkSLModifierFlags.cpp
+    src/sksl/ir/SkSLModifiersDeclaration.cpp
+    src/sksl/ir/SkSLPostfixExpression.cpp
+    src/sksl/ir/SkSLPrefixExpression.cpp
+    src/sksl/ir/SkSLProgram.cpp
+    src/sksl/ir/SkSLSetting.cpp
+    src/sksl/ir/SkSLStructDefinition.cpp
+    src/sksl/ir/SkSLSwitchCase.cpp
+    src/sksl/ir/SkSLSwitchStatement.cpp
+    src/sksl/ir/SkSLSwizzle.cpp
+    src/sksl/ir/SkSLSymbol.cpp
+    src/sksl/ir/SkSLSymbolTable.cpp
+    src/sksl/ir/SkSLTernaryExpression.cpp
+    src/sksl/ir/SkSLType.cpp
+    src/sksl/ir/SkSLTypeReference.cpp
+    src/sksl/ir/SkSLVarDeclarations.cpp
+    src/sksl/ir/SkSLVariable.cpp
+    src/sksl/ir/SkSLVariableReference.cpp
+    src/sksl/tracing/SkSLDebugTracePriv.cpp
+    src/sksl/tracing/SkSLTraceHook.cpp
+    src/sksl/transform/SkSLAddConstToVarModifiers.cpp
+    src/sksl/transform/SkSLEliminateDeadFunctions.cpp
+    src/sksl/transform/SkSLEliminateDeadGlobalVariables.cpp
+    src/sksl/transform/SkSLEliminateDeadLocalVariables.cpp
+    src/sksl/transform/SkSLEliminateEmptyStatements.cpp
+    src/sksl/transform/SkSLEliminateUnreachableCode.cpp
+    src/sksl/transform/SkSLFindAndDeclareBuiltinFunctions.cpp
+    src/sksl/transform/SkSLFindAndDeclareBuiltinStructs.cpp
+    src/sksl/transform/SkSLFindAndDeclareBuiltinVariables.cpp
+    src/sksl/transform/SkSLHoistSwitchVarDeclarationsAtTopLevel.cpp
+    src/sksl/transform/SkSLRenamePrivateSymbols.cpp
+    src/sksl/transform/SkSLReplaceConstVarsWithLiterals.cpp
+    src/sksl/transform/SkSLRewriteIndexedSwizzle.cpp
+
+    src/utils/SkCharToGlyphCache.cpp
+    src/utils/SkCustomTypeface.cpp
+    src/utils/SkDashPath.cpp
+    src/utils/SkMatrix22.cpp
+    src/utils/SkNWayCanvas.cpp
+    src/utils/SkPatchUtils.cpp
+    src/utils/SkPolyUtils.cpp
+    src/utils/SkShaderUtils.cpp
+    src/utils/SkShadowTessellator.cpp
+    src/utils/SkShadowUtils.cpp
+
+    src/opts/SkOpts_hsw.cpp
+    src/opts/SkOpts_skx.cpp
+
+    src/ports/SkDebug_stdio.cpp
+    src/ports/SkFontConfigInterface.cpp
+    src/ports/SkFontConfigInterface_direct.cpp
+    src/ports/SkFontConfigInterface_direct_factory.cpp
+    src/ports/SkFontHost_FreeType.cpp
+    src/ports/SkFontHost_FreeType_common.cpp
+    src/ports/SkFontMgr_fontconfig.cpp
+    src/ports/SkGlobalInitialization_default.cpp
+    src/ports/SkImageGenerator_skia.cpp
+    src/ports/SkMemory_malloc.cpp
+    src/ports/SkOSFile_posix.cpp
+    src/ports/SkOSFile_stdio.cpp
+
+    src/sfnt/SkOTTable_name.cpp
+    src/sfnt/SkOTUtils.cpp
+
+    modules/skcms/skcms.cc
+    modules/skcms/src/skcms_TransformBaseline.cc
+    modules/skcms/src/skcms_TransformHsw.cc
+    modules/skcms/src/skcms_TransformSkx.cc
+)
+
+find_package(Epoxy 1.5.4 REQUIRED)
+find_package(Freetype 2.9.0 REQUIRED)
+find_package(Fontconfig 2.13.0 REQUIRED)
+find_package(PNG REQUIRED)
```

Comment:
Again libraries already found so don't need to re-find them. Update the minimum version if necessary.

Reply:
This is a bit of a philosophical issue, as for example libpng is always needed regardless of which font/EGL/GL support libraries each port may want to use along with Skia, so I would argue at least the libpng one should be here, having the dependency right where it's used. OTOH we have the convention of writing down all the dependencies in the `Options<port>.cmake` files, but I feel that often the `find_package()` is “far” from where the dependency is used...

At any rate, let's go with the convention for now, I'll leave these be into `OptionsWPE.cmake` for now.

Reply:
I think the assumption should be that now that this is in `main` you're never going to build it outside WebKit and you should act accordingly. Its using some WebKit macros too.

If the required versions are tighter here than a normal WPE/GTK build then maybe the `Options${PORT}.cmake` should do something like

```
if (USE_SKIA AND FreeType_VERSION VERSION_LESS 2.9.0)
    message(FATAL_ERROR "FreeType version needs to be >= 2.9.0; Current version is ${FreeType_VERSION}")
endif ()
```

Reply:
So, my previous comment still applies, I would remove epoxy and PNG from here, keep the freetype and fontconfig, and move freetype and fontconfig inside a USE_CAIRO if in the main file.

Reply:
I think @carlosgcampos's proposal is a nice middle ground, I'll go with that.

Reply:
@donny-dont No, so far Skia doesn't require newer dependencies that versions we were already listing, so it's fine to pick the shared targets from `OptionsWPE.cmake` when they are already there 👌🏼